### PR TITLE
Design Picker: Redirect Partner Themes purchases to the Thank You page

### LINKED ISSUE #81604: Partner Themes on Onboarding: Atomic transfer state change
From https://github.com/Automattic/dotcom-forge/issues/2955#issuecomment-1711641530

---

While searching for other issues blocking this from launching, I've tested the flow again with the diff D118900-code applied to my sandbox:

- I visited `/start/business`
- Purchased a business plan
- Got redirected to `/setup/site-setup/goals?siteSlug=:site`, clicked continue
- Got redirected to `/setup/site-setup/designSetup?siteSlug=:site`. With the diff patched, the list contains partner themes as well, I selected the Makoney theme.
- Got redirected to `/setup/site-setup/designSetup?siteSlug=:site&theme=makoney`, I clicked the `Unlock this theme`, and got presented with this modal:
  CleanShot 2023-09-08 at 13 51 32@2x
- I clicked continue and completed the checkout
- Got redirected back to `/setup/site-setup/designSetup?siteSlug=:site&theme=makoney`, now the CTA reading `Continue`
  CleanShot 2023-09-08 at 13 52 23@2x
- I clicked continue, and got redirected to `/setup/site-setup/error`
  CleanShot 2023-09-08 at 13 56 52@2x

This prevents me from completing the onboarding flow. From SA I can see that the site was transferred to Atomic, and the theme purchase is also successful.

### PR DESCRIPTION
**Pull request by WBerredo**
Fixes https://github.com/Automattic/wp-calypso/issues/81604

## Proposed Changes

Redirect the user to the Thank You page after purchasing a Partner Theme during onboarding.

https://github.com/Automattic/wp-calypso/assets/5039531/350fa408-c58d-41f1-812b-55103fc7c28e

## Testing Instructions

* Apply the diff D118900-code to your sandbox and direct requests to the sandbox
* Go to the Design Picker page with a Marketplace theme selected `/setup/update-design/designSetup?siteSlug=:site&theme=:theme_slug`. Theme slugs such as `makoney` and `olsen-fse` can be used.
* Click on `Unlock Theme`
* You should see the modal with the updated layout.
* Click to proceed and complete the purchase
* After completing the purchase you should be redirected to the Thank You page to activate the theme.
* Test with atomic and non-atomic sites

CleanShot 2023-09-14 at 14 02 54@2x

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by epeicher — APPROVED** [review]
I have tested with:
- Atomic Business site
- Non-atomic Business site
- Premium (i.e. non-atomic) site

And all of the scenarios work as expected. Code changes also LGTM! :shipit:

**Comment by WBerredo** [comment]
@epeicher can we have a new round of review with the eligibility modal added now?

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx [hunk 1]
```diff
@@ -448,12 +449,19 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
...
 			plan = isEligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
 		}
+
+		// When the user is done with checkout, send them back to the current url
+		// If the theme is externally managed, send them to the marketplace thank you page
+		const destination = selectedDesign?.is_externally_managed
+			? addQueryArgs( `/marketplace/thank-you/${ siteSlug }`, {
+					themes: selectedDesign?.slug,
+			  } )
+			: window.location.href.replace( window.location.origin, '' );
```
Comment (epeicher) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:459]: Absolute minor comment, not even having a level of nitpick 😅. Could a line be added below to separate this piece of code from the call to `goToCheckout`? I initially thought it was incorrectly indented.

**Review by epeicher — APPROVED** [review]
I have tested it after the merge, and it works as expected. LGTM!

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx (hunk 1 above)
Comment (WBerredo) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:459]: done https://github.com/Automattic/wp-calypso/pull/81734/commits/31112e2e3b4eb289ab05b8136997d591e469525a

**Pull request merged**